### PR TITLE
Gracefully exit if the docker daemon is not running

### PR DESCRIPTION
**Pull request by raman-balyan**
This will check whether docker is running or not through the subprocess module. I have used the check_output method which is using "docker ps" command to check whether docker daemon is running or not. If docker is not running then subprocess module raises the ProcessError exception which is being used to raise the customized message on the console and gracefully exit the script.
Resolves: #28

Signed-off-by: Raman Balyan <raman.balyan@gmail.com>

### CODE REVIEW

File: report/report.py [hunk 1]
```diff
@@ -126,10 +127,14 @@ def generate_report(args, *images):
     write_report(report)
 
 
-
 def execute_dockerfile(args):
     '''Execution path if given a dockerfile'''
     logger.debug('Setting up...')
+    try:
+        subprocess.check_output('docker ps', shell=True)
```
Comment (nishakm) [inline report/report.py]: You could use the docker_command wrapper under utils/container.py. It doesn't use shell=True and takes care of things like logging and checking to see if the user doesn't have root privileges.

File: report/report.py [hunk 2]
```diff
@@ -126,10 +127,14 @@ def generate_report(args, *images):
...
+    except subprocess.CalledProcessError as error:
+        logger.error('Docker daemon is not running: {0}'.format(error.output))
```
Comment (nishakm) [inline report/report.py]: Is error.output a byte stream or a string?

File: report/report.py (hunk 1 above)
Comment (raman-balyan) [reply report/report.py]: thanks for the suggestion. I will update that..

File: report/report.py (hunk 2 above)
Comment (raman-balyan) [reply report/report.py]: It is a byte stream and I got my error :)

Comment (nishakm) [reply report/report.py]: I'm curious then - did you get the 'b' at the beginning of the error string? If yes then you will need to convert the byte stream into a string using error.output.decode('utf-8').

Comment (raman-balyan) [reply report/report.py]: Yes, I got 'b' and I will update it. Thanks

Comment (nishakm) [reply report/report.py]: No worries! Normally you would get an error saying you cannot convert a bytestring into a string but format does the !r conversion for you. It's nice to know exactly what types you're working with ^_^

Comment (nishakm) [reply report/report.py]: Yes, docker_command runs a shell command and raises a CalledProcessError is the command fails so you can put that in a try-except as you have done in report.py
>>> import subprocess
>>> from utils.container import docker_command
>>> try:
...     docker_command(['docker', 'ps'])
... except subprocess.CalledProcessError as error:
...     print(error.output.decode('utf-8'))
...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

**Comment by raman-balyan** [comment]
@nishakm I have updated the PR according to your comments. Also, I have updated the commit message with the description.

**Comment by raman-balyan** [comment]
@nishakm thanks for bearing with me. I will update that.

**Pull request closed without merging**